File: ios/ZulipMobile/UtilManager.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(UtilManager, NSObject)
  RCT_EXTERN_METHOD(randomBase64:(NSUInteger) length
                  resolver:(RCTPromiseResolveBlock *)resolve
                  rejecter:(RCTPromiseRejectBlock *)reject
                  )
@end

File: ios/ZulipMobile/UtilManager.swift
import Foundation

@objc(UtilManager)
class UtilManager: NSObject {
  @objc func randomBase64(_ length: Int, resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) -> Void {
    var keyData = Data(count: length)
    let result = keyData.withUnsafeMutableBytes {
      SecRandomCopyBytes(kSecRandomDefault, length, $0)
    }

    if result == errSecSuccess {
      resolve(keyData.base64EncodedString())
    } else {
      let error: NSError = NSError(domain: "zulip", code: Int(result), userInfo: nil)
      reject("random_failed", "Could not generate random data", error)
    }
  }
}
